Lib/annotationlib.py

```diff
@@ -154,8 +155,19 @@ def evaluate(self, *, globals=None, locals=None, type_params=None, owner=None):
                     globals[param_name] = param
                     locals.pop(param_name, None)
 
-        code = self.__forward_code__
-        value = eval(code, globals=globals, locals=locals)
+        arg = self.__forward_arg__
+        if arg.isidentifier():
+            if arg in locals:
+                value = locals[arg]
+            elif globals is not None and arg in globals:
```

Comment:
I don't think `globals` can be `None` here -- we make sure it's a `dict` no matter what on line 129 above